### PR TITLE
chore: use `queryAvailability` api data for calendar and lesson page

### PR DESCRIPTION
**Pull request by TheAlmightyCrumb**
It is a work in progress but connection to real data and a fake one is available for me & @DanielGovrin to be able to work with the UI for our corresponding tasks.

### CODE REVIEW

File: src/api/wix-api-context-provider.tsx [hunk 1]
```diff
@@ -66,32 +67,41 @@ function getWixApi(wixClient: ReturnType<typeof getWixClient>) {
...
+        cancelBooking: ({ _id, revision }: Pick<extendedBookings.Booking, '_id' | 'revision'>) =>
+            wixClient!.bookingsActions.cancelBooking(_id!, {
                 revision: revision!,
             }),
-
+        getServiceAvailability: (serviceId: string) => {
+            const startDate = new Date().toISOString();
+            const endDate = new Date(
+                new Date(startDate).setMonth(new Date(startDate).getMonth() + 2)
+            ).toISOString();
```
Comment (idango10) [inline src/api/wix-api-context-provider.tsx:78]: The `startDate` and `endDate` seem kinda arbitrary for me.

File: src/api/wix-api-context-provider.tsx [hunk 2]
```diff
@@ -66,32 +67,41 @@ function getWixApi(wixClient: ReturnType<typeof getWixClient>) {
...
+            //     //     return { success: false, url: '' };
+            //     // }
+            //     // const { redirectSession } = await wixClient.redirects.createRedirectSession({
+            //     //     ecomCheckout: { checkoutId },
+            //     //     callbacks: {
+            //     //         postFlowUrl: window.location.origin,
+            //     //         thankYouPageUrl: `${window.location.origin}${ROUTES.thankYou.to()}`,
+            //     //     },
+            //     // });
+            //     // return { success: true, url: redirectSession?.fullUrl };
```
Comment (idango10) [inline src/api/wix-api-context-provider.tsx:104]: Can revert the extra `//`'s here.

File: src/pages/lesson-page/lesson-page.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface LessonPageProps {
+    className?: string;
+}
+
+export const LessonPage = ({ className }: LessonPageProps) => {
+    const { slug } = useParams<RouteParams['/lesson/:slug']>();
+    const { data } = useLessonBySlug(slug);
+    const { data: availability, isLoading } = useAvailability(data?._id!);
+    console.log(availability);
```
Comment (idango10) [inline src/pages/lesson-page/lesson-page.tsx:16]: Remove `console.log`.

File: src/_codux/fake-data/fake-wix-api-context-provider.tsx [hunk 4]
```diff
@@ -35,6 +42,9 @@ function getWixApi(settings?: Settings): WixAPI {
             alert('Checkout');
             return Promise.resolve({ success: true, url: '' });
         },
+        getServiceAvailability: () => {
```
Comment (idango10) [inline src/_codux/fake-data/fake-wix-api-context-provider.tsx:45]: We probably want to pass `id` here, same as in `getLesson`.

**Review by idango10 — APPROVED** [review]

File: src/api/wix-api-context-provider.tsx (hunk 1 above)
Comment (TheAlmightyCrumb) [reply src/api/wix-api-context-provider.tsx:78]: I just put it to be the current date and 2 months forward for the sake of testing it, what defaults do you suggest? and is there a cleaner way than manipulating the `endDate` like I tried?

**Pull request merged**